source/byte_buf.c

```diff
@@ -1231,7 +1260,7 @@ bool aws_byte_buf_advance(
  */
 bool aws_byte_buf_write(struct aws_byte_buf *AWS_RESTRICT buf, const uint8_t *AWS_RESTRICT src, size_t len) {
     AWS_PRECONDITION(aws_byte_buf_is_valid(buf));
-    AWS_PRECONDITION(AWS_MEM_IS_WRITABLE(src, len), "Input array [src] must be readable up to [len] bytes.");
```

Comment:
NIT: I see that you're just fixing a bug but should we check the destination is writable?

Reply:
Assuming `AWS_PRECONDITION(aws_byte_buf_is_valid(buf));` is checking the validity of that argument